src/zenml/models/v2/core/pipeline_run.py

```diff
@@ -307,6 +309,62 @@ def produced_artifact_versions(self) -> List["ArtifactVersionResponse"]:
 
         return get_artifacts_versions_of_pipeline_run(self, only_produced=True)
 
+    def refresh_run_status(self) -> "PipelineRunResponse":
+        """Method to refresh the status of a pipeline run.
+
+        Returns:
+            The updated pipeline.
+
+        Raises:
+            ValueError: If the stack of the run response is None.
+        """
+        # Check if the stack still accessible
+        if self.stack is None:
+            raise ValueError(
+                "The stack that this pipeline run response was executed on"
+                "has been deleted."
+            )
+
+        # Create the orchestrator instance
+        from zenml.enums import StackComponentType
+        from zenml.orchestrators.base_orchestrator import BaseOrchestrator
+        from zenml.stack.stack_component import StackComponent
+
+        # Check if the stack still accessible
+        orchestrator_list = self.stack.components.get(
+            StackComponentType.ORCHESTRATOR, []
+        )
+        if len(orchestrator_list) == 0:
+            raise ValueError(
+                "The orchestrator that this pipeline run response was "
+                "executed with has been deleted."
+            )
+
+        orchestrator = cast(
+            BaseOrchestrator,
+            StackComponent.from_model(component_model=orchestrator_list[0]),
+        )
+
+        if self.status in [
```

Comment:
How about we move this up even above the orchestrator loading?

Reply:
On it.